### PR TITLE
Support for SSL API endpoint such as Infura (https://ipfs.infura.io/)

### PR DESCRIPTION
**Pull request by gjeanmart**
This is a very simple PR that allows java-ipfs-api to connect to third parties IPFS nodes secured over SSL.

### CODE REVIEW

**Comment by ianopolous** [comment]
We don't need any new constructors for this, as multiaddress can contain http or https. We do need to detect that though and use it internally.

**Comment by gjeanmart** [comment]
Hey,
Thanks for your answer. Understood how [[multiaddress](https://multiformats.io/multiaddr/)] works and agree it could be used for ssl detection rather than a flag.
For infura, I'm not sure what would be the the multiaddress: `/dns6/ipfs.infura.io/tcp/5001/https` or `/https/ipfs.infura.io/5001/https` 

Then, happy to make a change to detect `protocol` from multiaddress and consider `http` for any other constructors (host/port).

**Comment by ianopolous** [comment]
I believe it would be /dnsaddr/ipfs.infura.io/tcp/5001/https
(or dn6 or dn4 if you want those version of IP)

**Comment by gjeanmart** [comment]
Thanks for you answer. Will review my code and resubmit the PR.

File: src/main/java/io/ipfs/api/IPFS.java [hunk 1]
```diff
@@ -716,4 +724,8 @@ private Map postMap(String path, byte[] body, Map<String, String> headers) throw
             resp.write(buf, 0, r);
         return resp.toByteArray();
     }
+    
+    private static boolean detectSSL(MultiAddress multiaddress) {
+        return multiaddress.toString().contains("https");
```
[the remark this answers is not in the document]
Comment (gjeanmart) [reply src/main/java/io/ipfs/api/IPFS.java]: Thanks for your feedback. I guess we could detect the entire part `/https/`.

[the remark this answers is not in the document]
Comment (gjeanmart) [reply src/main/java/io/ipfs/api/IPFS.java]: I think the ssl detection could take part of MultiAddress for a better separation of concern:
```
    public boolean isHTTPOverSSL() {
        String[] parts = toString().substring(1).split("/");
        if (parts.length < 5)
            return false;
        if (!parts[4].equals("https"))
            return false;
        return true;
    } 
```
and then I would simply do:
```
public IPFS(MultiAddress addr) {
    this(addr.getHost(), addr.getTCPPort(), "/api/v0/", addr.isHTTPOverSSL());
}
```

However, I feel like the multiaddress standard so very underspecified so it's hard to imagine the right solution.

Let me know your thoughts.  
Best,
Greg

**Comment by margge** [comment]
@gjeanmart @ianopolous hi guys

I started using the library and I had the same issue. I am using Infura in my Android project.
I tested the solution proposed and it worked for me. 

I would like to know if those changes are going to be integrated.

**Pull request merged**